src/engine/Account-Imap.cpp

```diff
@@ -191,134 +190,70 @@ buildTokenInfo(const char *key, const GValue *value, gpointer data)
-    store in data, a (struct account_info*)
-    NOTE: this is a g_hash_table_foreach() function for a hash table of entries
-    key is a  pointer to the account guid, value is a gint32, 100000x
-    the probability for this account
-*/
-static void
-highestProbability(gpointer key, gpointer value, gpointer data)
-{
-    struct account_info *account_i = (struct account_info*)data;
-
-    /* if the current probability is greater than the stored, store the current */
-    if (GPOINTER_TO_INT(value) > account_i->probability)
+    account_probability (double p, double pd)
     {
-        /* Save the new highest probability and the assoaciated account guid */
-        account_i->probability = GPOINTER_TO_INT(value);
-        account_i->account_guid = (char*)key;
+        m_product = p;
+        m_product_difference = pd;
     }
-}
-
-
-//#define threshold (.90 * PROBABILITY_FACTOR) /* 90% */
+};
 
-static constexpr long int threshold = (.90 * PROBABILITY_FACTOR); // 90%
+/** conversion factors for the unordered map of 100000x the percentage match value,
+    ie. 10% would be  0.10 * 100000 = 10000
+ */
+static const double PROBABILITY_FACTOR = 100000.0;
+static const double CONFIDENCE {0.90}; // 90%
+static constexpr double threshold {CONFIDENCE * PROBABILITY_FACTOR};
 
 /** Look up an Account in the map */
 Account *
 GncImportMatchMap::find_account_bayes (GList* tokens)
 {
-    struct token_accounts_info tokenInfo; /**< holds the accounts and total
-                                           * token count for a single token */
-    GList *current_token;                 /**< pointer to the current
-                                           * token from the input GList
-                                           * tokens */
-    GList *current_account_token;         /**< pointer to the struct
-                                           * account_token_count */
-    struct account_token_count *account_c; /**< an account name and the number
-                                            * of times a token has appeared
-                                            * for the account */
-    struct account_probability *account_p; /**< intermediate storage of values
-                                            * to compute the bayes probability
-                                            * of an account */
-    GHashTable *running_probabilities = g_hash_table_new (g_str_hash, g_str_equal);
-    GHashTable *final_probabilities = g_hash_table_new (g_str_hash, g_str_equal);
-    struct account_info account_i;
+    GList *current_token;                 /**< pointer to the current token from the input GList tokens */
+
+    struct token_accounts_info tokenInfo; /**< holds the accounts and total token count for a single token */
+
+    struct account_token_count *account_c; /**< an account name and the number of times a token has appeared for the account */
+
+    GList *current_account_token;         /**< pointer to the struct account_token_count */
+
```

Comment:
In C++ we prefer to declare locals at the point of first use and prefer ```auto``` to an explicit type name if the local is initialized in a way that the compiler can derive the type (e.g. from the return type of a function). If a local will get its useful value inside a scope but used outside the scope, declare it right before the entry to that scope.